### PR TITLE
feat(fieldlabel)!: s2 migration

### PR DESCRIPTION
**Pull request by jawinn**
## Description

Spectrum 2 migration for the Field label component. The tokens used for several t-shirt sizes changed with S2, including for:
- Minimum height (component-height)
- Font size (increase for medium, large, and extra large)
- Top to text spacing
- Bottom to text spacing

These now match the S2 Figma design.

**Black and white static color variants**
Adds classes for static colors, as defined on the designs. And adds examples of these variants to the documentation. These use similar naming to the static variant classes used for Button:
- `.spectrum-FieldLabel--staticBlack`
- `.spectrum-FieldLabel--staticWhite`

Screenshot 2024-02-28 at 10 33 57 AM

**Deprecations and removals**
This also makes medium tokens the default and removes the redundant medium `.spectrum-FieldLabel--sizeM` class, to be in line with this convention followed during the Spectrum Tokens migrations.

Two mods already marked as deprecated for **Form** (its CSS is currently within Field label) have been removed.
Another mod that was referencing a global token was renamed: 
`--mod-disabled-content-color` has been renamed to `--mod-field-label-disabled-content-color`. 
Both have been noted in the updated Migration guide docs.

CSS-692

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Validation steps

- [x] Validation by design team @jawinn 
- [ ] Component tested in Storybook and Docs. Component spacing and font-size changes per t-shirt size to match the design spec.
- [ ] Component tested in other component examples where it is used in the docs, including the various form field components

### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:

- [ ] The pages render correctly, are accessible, and are responsive.

2. If components have been modified, VRTs have been run on this branch:

- [ ] VRTs have been run and looked at.
- [ ] Any VRT changes have been accepted (by reviewer and/or PR author), or there are no changes.

## To-do list

- [x] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [x] I have updated relevant storybook stories and templates.
- [x] I have tested these changes in Windows High Contrast mode.
- [x] If my change impacts **other components**, I have tested to make sure they don't break.
- [x] If my change impacts **documentation**, I have updated the documentation accordingly.
- [ ] ✨ This pull request is ready to merge. ✨

### CODE REVIEW

File: components/fieldlabel/index.css [hunk 1]
```diff
@@ -116,19 +106,25 @@ governing permissions and limitations under the License.
...
   --spectrum-tableform-item-block-spacing: var(--spectrum-spacing-300);
 
-  margin: 0;
   display: grid;
   grid-template-columns: var(--mod-form-grid-template-columns, auto auto);
-  inline-size: var(--mod-form-inline-size, fit-content);
+  row-gap: var(--mod-form-item-block-spacing, var(--spectrum-tableform-item-block-spacing));
   justify-content: start;
 
-  /* @deprecation --mod-tableform-item-block-spacing has been renamed to
```
Comment (mdt2) [inline components/fieldlabel/index.css:129]: Thanks for removing these, seems like the perfect time to since this is a breaking change ✨

**Review by mdt2 — COMMENTED** [review]
Really appreciate the thorough PR description 👏 

Tokens in the code are mostly aligned with the design spec. I see the text-to-asterisk tokens in the css, but are we missing the top-to-asterisk tokens? Or is that somehow handled with the vertical-align on `.spectrum-FieldLabel-requiredIcon`?

**Comment by jawinn** [comment]
> I see the text-to-asterisk tokens in the css, but are we missing the top-to-asterisk tokens? Or is that somehow handled with the vertical-align on `.spectrum-FieldLabel-requiredIcon`?

It isn't necessary for us to use that token. Its alignment is handled by the `baseline` vertical-alignment, yeah. I handled some alignment updates on this component in #2166 and there are some more details in that PR description when it was confirmed with design---it was connected to some Figma limitations vs how we implement it.

File: components/fieldlabel/index.css [hunk 2]
```diff
@@ -170,17 +164,17 @@ governing permissions and limitations under the License.
...
+      color: var(--highcontrast-field-label-disabled-content-color, var(--mod-field-label-disabled-content-color, var(--spectrum-disabled-content-color)));
     }
   }
 }
 
 /********* WHCM *********/
 @media (forced-colors: active) {
   .spectrum-FieldLabel {
-    --highcontrast-disabled-content-color: GrayText;
+    --highcontrast-field-label-disabled-content-color: GrayText;
```
Comment (pfulton) [inline components/fieldlabel/index.css:178]: Great catch.

**Review by pfulton — APPROVED** [review]
This looks good to me. Thanks for being so thorough in the PR description and taking care to write the docs in a way that explains the changes!

I have one outstanding question for SWC integration + making `medium` the default: will they need to make any DOM changes on their end? If so, we _might_ need to either wait to make that change OR, if possible, we try to land that change in S1 so they can do it _now_.

I'll follow up with them.

**Comment by pfulton** [comment]
Chromatic review available here: https://www.chromatic.com/review?appId=64762974a45b8bc5ca1705a2&number=6&type=unlinked

**Pull request merged**